frontend/src/api_methods.ts

```diff
@@ -11,4 +11,9 @@ export function getStarterPageMovies() {
 export function getStreamingDetails(movieName: string) {
     return axios.get("api/soundtracker/streaming/" + movieName)
         .then((response: AxiosResponse<StreamingStatusDTO>) => response.data)
+}
+
+export function searchForMovie(query: string) {
+    return axios.get("api/soundtracker/search/" + query)
+        .then(response => response.data)
```

Comment:
I would add a type here for `response`, so that TypeScript knows what `data` is.